web3/main.py

```diff
@@ -285,7 +296,34 @@ def pm(self) -> "PM":
     def enable_unstable_package_management_api(self) -> None:
         from web3.pm import PM  # noqa: F811
         if not hasattr(self, '_pm'):
-            PM.attach(self, '_pm')
+            self.attach_modules({'_pm': (PM,)})
 
     def enable_strict_bytes_type_checking(self) -> None:
         self.codec = ABICodec(build_strict_registry())
+
+    def attach_modules(
+        self,
+        module_definitions: Dict[str, Sequence[Any]],
+        parent: Optional[T] = None
```

Comment:
This typing looks suspect.  Typically `TypeVar` is used to say something like "this function takes a value of type `T` and returns a `Sequence[T]`".  Roughly speaking, it's for establishing a linkage between different arguments and return values.  On it's own, I don't know exactly how it behaves, but I think this is effectively an `Any` type.

Reply:
Oh thanks! That makes sense. It should actually either be a Web3 instance or a Module, so I'll update that!